### ISSUE TITLE
Script ends with php8.2.0 not compatible

### ISSUE THREAD

**Issue by ikorbln** [nextcloud/nextcloudpi#1675]
Hi,

I install a fresh system with debian 11.5 on a intel system.
After the initial install of debian, i checked php. No version is installed.

After running the ncp install script it ends after some time with:
"This version of Nextcloud is not compatible with PHP>=8.2.<br/>You are currently running 8.2.0".

I can see that the script installs 8.1 and 8.2.
Do i miss something?

**System information**

<details>
<summary>NextCloudPi diagnostics</summary>

```

grep: /var/www/nextcloud/config/config.php: Datei oder Verzeichnis nicht gefunden
NextCloudPi version  v1.50.3
NextCloudPi image    NextCloudPi_12-13-22
OS                   Debian GNU/Linux 11. 5.10.0-19-amd64 (x86_64)
automount            no
USB devices          none
datadir               (doesn't exist)
rootfs usage         4,6G/250G
swapfile             /var/swap
dbdir                /var/lib/mysql
Nextcloud check      error
HTTPD service        up
PHP service          up
MariaDB service      up
Redis service        up
HPB service          down
Postfix service      up
Internet check       ok
public IP            ***REMOVED SENSITIVE VALUE***
public IP            ***REMOVED SENSITIVE VALUE***
Port check 80        closed
Port check 443       closed
IP                   ***REMOVED SENSITIVE VALUE***
gateway              ***REMOVED SENSITIVE VALUE***
Interface            ens18
certificates         ***REMOVED SENSITIVE VALUE***
NAT loopback         no
Uptime               22min

```
</details>

<details>
<summary>Nextcloud configuration</summary>

```

This version of Nextcloud is not compatible with PHP>=8.2.<br/>You are currently running 8.2.0.
```
</details>

<details>
<summary>HTTPd logs</summary>

```

[Tue Dec 13 16:44:13.986178 2022] [mpm_event:notice] [pid 2708:tid 140453277723968] AH00489: Apache/2.4.54 (Debian) configured -- resuming normal operations
[Tue Dec 13 16:44:13.986315 2022] [core:notice] [pid 2708:tid 140453277723968] AH00094: Command line: '/usr/sbin/apache2'
[Tue Dec 13 16:48:46.750183 2022] [mpm_event:notice] [pid 2708:tid 140453277723968] AH00492: caught SIGWINCH, shutting down gracefully
[Tue Dec 13 16:48:46.806690 2022] [mpm_event:notice] [pid 30043:tid 139812760730944] AH00489: Apache/2.4.54 (Debian) OpenSSL/1.1.1n configured -- resuming normal operations
[Tue Dec 13 16:48:46.807247 2022] [core:notice] [pid 30043:tid 139812760730944] AH00094: Command line: '/usr/sbin/apache2'

```
</details>

<details>
<summary>Database logs</summary>

```

tail: '/var/log/mysql/*.log' kann nicht zum Lesen geöffnet werden: Datei oder Verzeichnis nicht gefunden

```
</details>

grep: /var/www/nextcloud/config/config.php: Datei oder Verzeichnis nicht gefunden

<details>
<summary>Nextcloud logs</summary>

```

tail: '/nextcloud.log' kann nicht zum Lesen geöffnet werden: Datei oder Verzeichnis nicht gefunden

```
</details>

**Comment by ikorbln** [comment]
Littel Update: i added the source https://packages.sury.org/php/ and set the php8.2 on hold.
Now the script ends with this error:
php-json : depends on: php8.2-json
E: Problems cannot be corrected, you have held defective packages.

So there is something with this php setup-part in the script.

**Comment by ikorbln** [comment]
Hi again,
A user in the german forum found the "solution" for this.
Remove the php-json in Line 50 in the lamp.sh.
Now the script installs without error and the nextcloud is full working.

**Comment by tteck** [comment]
replace `php-json` with `php${PHPVER}-common`

**Comment by ikorbln** [comment]
> replace `php-json` with `php${PHPVER}-common`

this pkg was installed anyway.

**Comment by michiey** [comment]
Hi,
I have the same problem. Neither removing php-json from line 50 in lamp.sh nor replacing it with php${PHPVER}-common helped.
Installation fails with
"This version of Nextcloud is not compatible with PHP>=8.2.
You are currently running 8.2.0"

Fresh und basic debian bullseye LXC, no php installed

**Comment by ikorbln** [comment]
> Hi, I have the same problem. Neither removing php-json from line 50 in lamp.sh nor replacing it with php${PHPVER}-common helped. Installation fails with "This version of Nextcloud is not compatible with PHP>=8.2. You are currently running 8.2.0"
> 
> Fresh und basic debian bullseye LXC, no php installed

Hi,
You must do one thing extra.
You have to clone the repo lokal (i do it with root).
So under /root ther is a folder nextcloudpi. There you do the changes to the lamp.sh. But you must modifiy the install.sh to.
Because it loads the repo again to /tmp and uses the unmodified version.
So, do the changes in the install.sh and insert after line 38:
`cp -r /root/nextcloudpi "${CODE_DIR}"`
and coment out the git clone line

the run the install.sh (in my example it is under /root/nextcloudpi)

**Comment by tteck** [comment]
Try 
```
curl -sSL https://raw.githubusercontent.com/tteck/nextcloudpi/master/install.sh | bash
```

**Comment by michiey** [comment]
@tteck @ikorbln thank you both, it worked

**Comment by theCalcaholic** [comment]
Thank you all. The installer should now be working again.
